R/mediate.R

```diff
@@ -0,0 +1,394 @@
+#' Perform mediation analysis
+#'
+#' \code{getMediation} provides a wrapper of \code{\link[mediation:mediate]{mediate}}
+#' for \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}.
+#'
+#' @param x a \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}.
+#' 
+#' @param outcome The name of the colData variable used as outcome in the model.
+#' 
+#' @param treatment The name of the colData variable used as treatment in the model.
+#'
+#' @param mediator The name of the colData variable used as mediator in the model.
+#'   (default: \code{mediator = NULL})
+#'
+#' @param assay.type The name of the assay used for feature-wise mediation analysis.
+#'   (default: \code{assay.type = NULL})
+#' 
+#' @param dim.type The name of the reducedDim used for component-wise mediation analysis.
+#'   (default: \code{dim.type = NULL})
+#'
+#' @param family A specification for the outcome model link function.
+#'    (default: \code{family = gaussian("identity")})
+#' 
+#' @param covariates Name or list of names of colData variables used as covariates
+#'   in the model. (default: \code{covariates = NULL})
+#' 
+#' @param p.adj.method A single character value for selecting adjustment method
+#'   of p-values. Passed to `p.adjust` function. (default: \code{p.adj.method = "BH"})
```

Comment:
default in p.adjust function is "holm" I think (can be checked). I think we should not change the default if the argument is passed. Instead we can provide our own recommendation in the example section and show how to use.

Reply:
Yes, default of p.adjust is "holm". I saw "fdr"/"BH" is the default of `testExperimentCrossCorrelation`, so I used it. So should we use "holm"?

Reply:
Ah right. It would be good for us to be consistent within mia at least. 

In my opinion it would be most clean to do as I proposed. We can have "BH"/"fdr" in all examples. But default can be invisible the user, then it is confusing if it changes the default value of p.adjust while we at the same time say that the function calls that.

Reply:
Btw isn't it it nowadays the real name `testExperimentCrossAssociation`?